packages/alfa-rules/src/sia-r69/rule.ts

```diff
@@ -0,0 +1,388 @@
+import { Rule } from "@siteimprove/alfa-act";
+import { Role } from "@siteimprove/alfa-aria";
+import { RGB, Percentage, Current, System } from "@siteimprove/alfa-css";
+import { Device } from "@siteimprove/alfa-device";
+import { Element, Text, Namespace, Node } from "@siteimprove/alfa-dom";
+import { Iterable } from "@siteimprove/alfa-iterable";
+import { Option, None } from "@siteimprove/alfa-option";
+import { Predicate } from "@siteimprove/alfa-predicate";
+import { Ok, Err } from "@siteimprove/alfa-result";
+import { Style } from "@siteimprove/alfa-style";
+import { Page } from "@siteimprove/alfa-web";
+
+import { expectation } from "../common/expectation";
+
+import { hasCategory } from "../common/predicate/has-category";
+import { hasRole } from "../common/predicate/has-role";
+import { isDisabled } from "../common/predicate/is-disabled";
+import { isIgnored } from "../common/predicate/is-ignored";
+import { isVisible } from "../common/predicate/is-visible";
+
+import { Question } from "../common/question";
+
+const { reduce, some, flatMap, map, concat } = Iterable;
+const { and, not, equals, test } = Predicate;
+const { min, max } = Math;
+
+export default Rule.Atomic.of<Page, Text, Question>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r69.html",
+  evaluate({ device, document }) {
+    return {
+      applicability() {
+        return visit(document);
+
+        function* visit(node: Node): Iterable<Text> {
+          if (Element.isElement(node)) {
```

Comment:
This (and https://github.com/Siteimprove/alfa/pull/228/files#diff-3dbf3df10d60073193da7fa74c7f22fcR48) make me think we maybe should have an `node#kind` or something because
```javascript
switch (node.kind) {
  case "element": …
  case "text": …
}
```
might be more readable.

Reply:
I do agree that would be ideal. The issue is that we can't narrow `Node` based on a property as it's not a union type; the DOM types are modelled using a class hierarchy.